File: Frontend/src/types/brief.ts
import type { JourneyStatus, TourFreshness, TourStep } from "./tour";

export type BriefWarning = { kind: string; message: string; url: string | null };

export type BriefPreview = {
  issueUrl: string;
  repoName: string;
  issueNumber: number;
  title: string;
  state: string;
  labels: string[];
  assignees: string[];
  warnings: BriefWarning[];
  targetBranch: {
    branch: string | null;
    source: string;
    evidence: string | null;
    evidencePath: string | null;
    defaultBranch: string | null;
  };
  forkStatus: {
    forkRepo: string | null;
    upstreamRepo: string;
    commitsBehind: number | null;
    measurable: boolean;
  };
};

export type BriefArtifact = {
  issue_title: string;
  issue_number: number;
  repo_name: string;
  summary: string;
  house_rules: string[];
  setup_recipe: {
    steps: string[];
    target_branch: string | null;
    target_branch_source: string;
    target_branch_evidence: string | null;
    target_branch_evidence_path: string | null;
    default_branch: string | null;
    fork_repo: string | null;
    upstream_repo: string;
    fork_commits_behind: number | null;
    fork_status_measurable: boolean;
  };
  reading_steps: TourStep[];
  test_guidance: string[];
  plan_checklist: string[];
  freshness: TourFreshness | null;
  preflight_warnings: BriefWarning[];
  confidence: {
    level: string;
    issue_is_vague: boolean;
    questions_for_maintainer: string[];
  };
  honesty_note: string | null;
};

export type BriefResponse = {
  id: number;
  status: JourneyStatus;
  phase: "blocked_on_ingest" | "queued" | "generating" | JourneyStatus;
  repoName: string;
  ref: string | null;
  issueNumber: number;
  issueTitle: string;
  artifact: BriefArtifact | null;
  error: string | null;
};

export type BriefSummary = Pick<
  BriefResponse,
  "id" | "status" | "repoName" | "ref" | "issueNumber" | "issueTitle"
> & { createdAt: string };

File: Backend/app/models/brief.py
"""Structured, user-facing issue brief artifact."""

from pydantic import BaseModel, Field

from app.models.tour import TourFreshness, TourStep


class BriefWarning(BaseModel):
    kind: str
    message: str = Field(min_length=1)
    url: str | None = None


class BriefConfidence(BaseModel):
    level: str
    issue_is_vague: bool = False
    questions_for_maintainer: list[str] = Field(default_factory=list)


class BriefSetupRecipe(BaseModel):
    steps: list[str] = Field(default_factory=list)
    target_branch: str | None = None
    target_branch_source: str
    target_branch_evidence: str | None = None
    target_branch_evidence_path: str | None = None
    default_branch: str | None = None
    fork_repo: str | None = None
    upstream_repo: str
    fork_commits_behind: int | None = Field(default=None, ge=0)
    fork_status_measurable: bool = False


class BriefArtifact(BaseModel):
    issue_title: str = Field(min_length=1)
    issue_number: int = Field(ge=1)
    repo_name: str = Field(min_length=1)
    summary: str = Field(min_length=1)
    house_rules: list[str] = Field(default_factory=list)
    setup_recipe: BriefSetupRecipe
    reading_steps: list[TourStep] = Field(default_factory=list)
    test_guidance: list[str] = Field(default_factory=list)
    plan_checklist: list[str] = Field(default_factory=list)
    freshness: TourFreshness | None = None
    preflight_warnings: list[BriefWarning] = Field(default_factory=list)
    confidence: BriefConfidence
    honesty_note: str | None = None
